api/backup.go

```diff
@@ -18,7 +19,61 @@ func (s *Server) BackupTargetList(w http.ResponseWriter, req *http.Request) erro
 	if err != nil {
 		return errors.Wrap(err, "failed to list backup targets")
 	}
-	apiContext.Write(toBackupTargetCollection(backupTargets))
+	apiContext.Write(toBackupTargetCollection(backupTargets, apiContext))
+	return nil
+}
+
+func (s *Server) BackupTargetUpdateAll(w http.ResponseWriter, req *http.Request) error {
+	var input BackupTarget
+
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&input); err != nil {
+		return err
+	}
+
+	bts, err := s.m.ListBackupTargetsSorted()
+	if err != nil {
+		return errors.Wrap(err, "failed to list backup targets")
+	}
+
+	if input.SyncBackupTarget {
+		for _, bt := range bts {
+			if _, err := s.m.SyncBackupTarget(bt); err != nil {
+				logrus.WithError(err).Warnf("Failed to synchronize backup target %v", bt.Name)
+			}
+		}
+	}
+
+	apiContext.Write(toBackupTargetCollection(bts, apiContext))
+	return nil
+}
+
+func (s *Server) SyncBackupTarget(w http.ResponseWriter, req *http.Request) error {
+	var input SyncBackupTarget
```

Comment:
Why not just use `BackupTarget` model directly? just want the `name`?

Reply:
Moved to struct `SyncBackupResource`.
Thanks.